internal/plugintest/terraform_json_buffer_test.go

```diff
@@ -0,0 +1,217 @@
+package plugintest
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+func TestTerraformJSONDiagnostics_Contains(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		diags    []tfjson.Diagnostic
+		regex    *regexp.Regexp
+		severity tfjson.DiagnosticSeverity
+		expected bool
+	}{
+		"severity-not-found": {
+			diags: []tfjson.Diagnostic{
+				{
+					Severity: tfjson.DiagnosticSeverityError,
+					Summary:  "error summary",
+					Detail:   "error detail",
+				},
+			},
+			regex:    regexp.MustCompile("error summary"),
+			severity: tfjson.DiagnosticSeverityWarning,
+			expected: false,
+		},
+		"summary-not-found": {
+			diags: []tfjson.Diagnostic{
+				{
+					Severity: tfjson.DiagnosticSeverityError,
+					Summary:  "warning summary",
+					Detail:   "warning detail",
+				},
+			},
+			regex:    regexp.MustCompile("error detail"),
+			severity: tfjson.DiagnosticSeverityError,
+			expected: false,
+		},
+		"summary-found": {
+			diags: []tfjson.Diagnostic{
+				{
+					Severity: tfjson.DiagnosticSeverityError,
+					Summary:  "error summary",
+					Detail:   "error detail",
+				},
+			},
+			regex:    regexp.MustCompile("error summary"),
+			severity: tfjson.DiagnosticSeverityError,
+			expected: true,
+		},
+		"detail-found": {
+			diags: []tfjson.Diagnostic{
+				{
+					Severity: tfjson.DiagnosticSeverityError,
+					Summary:  "error summary",
+					Detail:   "error detail",
+				},
+			},
+			regex:    regexp.MustCompile("error detail"),
+			severity: tfjson.DiagnosticSeverityError,
+			expected: true,
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+
+		t.Run(name, func(t *testing.T) {
+			var tfJSONDiagnostics TerraformJSONDiagnostics = testCase.diags
+
+			isFound := tfJSONDiagnostics.Contains(testCase.regex, testCase.severity)
+
+			if diff := cmp.Diff(isFound, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestTerraformJSONBuffer_Parse(t *testing.T) {
+	t.Parallel()
+
+	tfJSON := NewTerraformJSONBuffer()
+
+	file, err := os.Open("../testdata/terraform-json-output.txt")
+	if err != nil {
+		t.Errorf("cannot read file: %s", err)
+	}
+	defer file.Close()
+
+	var fileEntries []string
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		txt := scanner.Text()
+
+		_, err := tfJSON.Write([]byte(txt + "\n"))
+		if err != nil {
+			t.Errorf("cannot write to tfJSON: %s", err)
+		}
+
+		fileEntries = append(fileEntries, txt)
+	}
+
+	if err := scanner.Err(); err != nil {
+		t.Errorf("scanner error: %s", err)
+	}
```

Comment:
Since `os.File` is itself an `io.Reader`, can it just be copied to `tfJSON`? e.g.

```go
	tfJSON := NewTerraformJSONBuffer()

	file, err := os.Open("../testdata/terraform-json-output.txt")
	if err != nil {
		t.Errorf("cannot read file: %s", err)
	}
	defer file.Close()

	_, err := io.Copy(tfJSON, file)

	if err != nil {
		t.Fatalf("cannot copy file contents to buffer: %s", err)
	}
```

I think then we could take that a step further if we wanted so you could potentially create the `TerraformJSONBuffer` using file contents for when we get more test cases:

```go
func NewTerraformJSONBufferFromFile(path string) error {
	tfJSON := NewTerraformJSONBuffer()

	file, err := os.Open(path)

	if err != nil {
		return fmt.Errorf("cannot read file: %w", err)
	}

	defer file.Close()

	_, err := io.Copy(tfJSON, file)

	if err != nil {
		return fmt.Errorf("cannot copy file contents to buffer: %w", err)
	}

	return nil
}
```

Or conversely it could just be a method such as `ReadFile()`:

```go
func (b TerraformJSONBuffer) ReadFile(path string) error {
	file, err := os.Open(path)

	if err != nil {
		return fmt.Errorf("cannot read file: %w", err)
	}

	defer file.Close()

	_, err := io.Copy(b, file)

	if err != nil {
		return fmt.Errorf("cannot copy file contents to buffer: %w", err)
	}

	return nil
}
```

Reply:
Have incorporated both `NewTerraformJSONBufferFromFile` and `ReadFile`. Thanks!